fem/intrules.hpp

```diff
@@ -84,6 +84,17 @@ class IntegrationPoint
    void Set1w(const double x1, const double w) { x = x1; weight = w; }
 
    void Set1w(const double *p) { x = p[0]; weight = p[1]; }
+
+   bool operator==(const IntegrationPoint &rhs) const
+   {
+      return x==rhs.x && y==rhs.y && z==rhs.z && weight==rhs.weight &&
+             index==rhs.index;
+   }
```

Comment:
Just to note, one possible pitfall here is that integration points in 2D may have uninitialized data in the `z` coordinate (and the same goes for integration points in 1D with the `y` coordinate).

Reply:
Oh, good catch, we should probably be initializing them to 0.0 at the very least.  This would lead to equality in some situations between 1D, 2D and 3D integration points.